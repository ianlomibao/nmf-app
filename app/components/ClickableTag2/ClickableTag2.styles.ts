import { StyleSheet } from "react-native";

import { Colors } from "style";

export default StyleSheet.create({
    textBox: {
        padding: 10,
        borderRadius: 5,
        margin: 10,
        borderWidth: 2,
        
    },
    selected: {
        backgroundColor: Colors.green10,
        borderColor: Colors.green50,
    },
    unselected: {
        backgroundColor: Colors.grey10,
        borderColor: Colors.grey10,
    },
    text: {
        fontSize: 16,
        textAlign: "center",
        fontWeight: "bold",
    },
    selectedText: {
        color: Colors.green50,
    },
    unselectedText: {
        color: Colors.grey70,
    },
});