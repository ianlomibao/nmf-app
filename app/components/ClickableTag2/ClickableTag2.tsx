import React from "react";
import { View } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";

import Text from "../Text";
import styles from "./ClickableTag2.styles";

type Props = {
    text: string;
    isSelected: boolean;
    onPress: () => void;
};

const ClickableTag2: React.FC<Props> = ({text, isSelected, onPress}) => (
    <TouchableOpacity onPress={onPress}>
        <View style={[styles.textBox, isSelected ? styles.selected : styles.unselected ]}>
            <Text.Secondary style={[styles.text, isSelected ? styles.selectedText : styles.unselectedText ]}>{text}</Text.Secondary>
        </View>
    </TouchableOpacity>
);

export default ClickableTag2;