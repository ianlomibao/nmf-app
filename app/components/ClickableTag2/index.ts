export {default} from "./ClickableTag2";