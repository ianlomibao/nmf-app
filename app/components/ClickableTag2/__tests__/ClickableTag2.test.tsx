import React from "react";
import { create } from "react-test-renderer";

import ClickableTag2 from "../";

jest.unmock("../");

const foo = jest.fn();

it("renders selected ClickableTag", () => {
    const tree = create(<ClickableTag2 text={"Selected"} isSelected={true} onPress={foo}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders unselected ClickableTag", () => {
    const tree = create(<ClickableTag2 text={"Unselected"} isSelected={false} onPress={foo}/>).toJSON();
    expect(tree).toMatchSnapshot();
});