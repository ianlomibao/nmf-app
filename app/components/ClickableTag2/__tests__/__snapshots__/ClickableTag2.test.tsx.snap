// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders selected ClickableTag 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 5,
            "borderWidth": 2,
            "margin": 10,
            "padding": 10,
          },
          Object {
            "backgroundColor": "#DCF0E5",
            "borderColor": "#4BB377",
          },
        ]
      }
    >
      <Text.Secondary
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {
              "color": "#4BB377",
            },
          ]
        }
      >
        Selected
      </Text.Secondary>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`renders unselected ClickableTag 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  exclusive={true}
  handlerTag={2}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 5,
            "borderWidth": 2,
            "margin": 10,
            "padding": 10,
          },
          Object {
            "backgroundColor": "#F0F0F0",
            "borderColor": "#F0F0F0",
          },
        ]
      }
    >
      <Text.Secondary
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {
              "color": "#595D58",
            },
          ]
        }
      >
        Unselected
      </Text.Secondary>
    </View>
  </View>
</RNGestureHandlerButton>
`;
